src/tools/genpolicy/src/mount_and_storage.rs

```diff
@@ -107,6 +107,23 @@ pub fn get_mount_and_storage(
         get_empty_dir_mount_and_storage(settings, p_mounts, storages, yaml_mount, memory_medium);
     } else if yaml_volume.persistentVolumeClaim.is_some() || yaml_volume.azureFile.is_some() {
         get_shared_bind_mount(yaml_mount, p_mounts, "rprivate", "rw");
+
+        if yaml_volume.persistentVolumeClaim.is_some() {
+            // Add a storage for the potential virtio-blk device.
+            storages.push(agent::Storage {
+                driver: "blk".into(),
+                driver_options: Vec::new(),
+                fs_group: None,
+                // source and mount_point can't be generated as those are
+                // computed by the shim as follows:
+                //   source = PCI device ID (e.g. "02")
+                //   mount_point = "/run/kata-containers/sandbox/storage/$(base64_encode(source))"
+                source: "".into(),
```

Comment:
It sems like the source should be similar to "$(pci-device-id)" - similar to the current "$(node-name)" from rules.rego.

Then, either the Agent of rules.rego would decode the incoming pci-device-id and send it to the Rego rules. Not sure how easy is to base64 decode in Rego (including in the upcoming Regorus), so the Agent is typically a better place to decode.

Then, the Rego rules would verify the format of the pci-device-id - e.g., by using some regex.

Reply:
Not sure how we'd pass pci-device-id from the agent to Rego? Do we do that with anything else at the moment?

Reply:
Example of "decoding" data before sending it to Rego: https://github.com/microsoft/kata-containers/commit/d3181c84dbee2e0388700df21ec61e675f86f775

In that example, the decoding step is:

PathBuf::from(OsStr::from_bytes(&req.data))